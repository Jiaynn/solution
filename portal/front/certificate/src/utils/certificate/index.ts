/*
 * @file component certificate utils
 * @author Yao Jingtian <yncst233@gmail.com>
 */

import { sortBy, last, without, uniq, values } from 'lodash'

import { OrderType } from '../../apis/ssl'
import {
  sslMap, canBeClosedStatuses, SSLDomainType, CertType,
  OrderStatus, ProductShortName, SslBrand, isDVSSLType
} from '../../constants/ssl'
import { isOEM } from '../../constants/env'
import { isStatusDone, isStatusPending } from './status'

interface ICertBrandInfo {
  brand: SslBrand
  certType: CertType
}

// 根据 productShortName 获取证书品牌、类型
export function shortNameToInfo(productShortName?: ProductShortName): ICertBrandInfo {
  let brand: SslBrand
  let certType: CertType
  Object.keys(sslMap).some((sslBrand: SslBrand) => Object.keys(sslMap[sslBrand]).some((sslType: CertType) => {
    if (sslMap[sslBrand][sslType]!.shortName === productShortName) {
      brand = sslBrand
      certType = sslType
      return true
    }
    if (productShortName && sslMap[sslBrand][sslType]!.domainShortNameMap) {
      const names = values(sslMap[sslBrand][sslType]!.domainShortNameMap)
      if (names.indexOf(productShortName) > -1) {
        brand = sslBrand
        certType = sslType
        return true
      }
    }
    return false
  }))
  return {
    brand: brand!,
    certType: certType!
  }
}

export function getLatestDnsNames(replaceOrders: any[], dnsNames?: string[]): string[] {
  if (!replaceOrders || replaceOrders.length === 0) {
    return dnsNames || []
  }
  const list = sortBy(replaceOrders, 'create_time')
    .filter(order => order.state === 1)
  if (list.length > 0) {
    return last(list).dnsnames.split(',')
  }
  return dnsNames || []
}

export function hasRepeatItem(arr: any[], compareArr?: any[]) {
  if (arr && compareArr) {
    return without(arr.slice(), ...compareArr.slice()).length !== arr.length
  }
  return uniq(arr.slice()).length !== arr.length
}

/**
 * 从后端接口字段 renewable 来判断
 * 历史前端逻辑：已签发/已重颁发 过期前 30 天的非重颁发订单，显示【续费】(若 notAfter <= 0，则根据 providerRenewable 字段)
 */
export function canRenew(renewable: boolean) {
  return isOEM ? false : renewable
}

interface ICanAddDomainParams {
  notAfter: number,
  state: number,
  productType: SSLDomainType,
  orderType: OrderType
}

// 已签发/已重颁发 非重颁发 未过期的多域名、多域名泛域名订单，显示【添加域名】
export function canAddDomain(params: ICanAddDomainParams) {
  const { notAfter, state, productType, orderType } = params
  const now = Date.now()
  if (!state || !isStatusDone(state)) {
    return false
  }
  if ([SSLDomainType.Multiple, SSLDomainType.MultipleWildcard].indexOf(productType) === -1) {
    return false
  }
  if (orderType === OrderType.Replace) {
    return false
  }
  if (!notAfter) {
    return false
  }
  return (
    notAfter <= 0
    || now < notAfter * 1000
  )
}

interface ICanUploadConfirmationParams {
  state: number,
  productShortName: ProductShortName,
  uploadConfirmLetter: boolean
}

// 非 DV 且待确认状态才显示【上传确认函】
export function canUploadConfirmation(params: ICanUploadConfirmationParams) {
  const { state, productShortName, uploadConfirmLetter } = params
  if (!state || !isStatusPending(state)) {
    return false
  }

  const sslInfo = shortNameToInfo(productShortName)

  return !uploadConfirmLetter && !isDVSSLType(sslInfo.certType)
}

export function canClose(state: number): boolean {
  return canBeClosedStatuses.indexOf(state) !== -1
}

// 过期 7 天前的证书都能够部署
export function canCertDeploy(notAfter: number): boolean {
  return !isOEM && ((notAfter - 7 * 24 * 3600) > (Date.now() / 1000))
}

// 判断证书类型是否为私有证书
export function isPrivateCert(productShortName: ProductShortName) {
  return !productShortName
}

// 已经完成的订单，并且满足证书部署条件的订单能够部署
export function canOrderDeploy(notAfter: number, state: OrderStatus): boolean {
  return isStatusDone(state) && canCertDeploy(notAfter)
}

// 2 年的续期证书需要补充文案
export function getCertYearTipsByYear(year: number): string {
  return year === 2 ? '续期证书' : ''
}

// 自动续期的证书、订单需要补充文案
export function getCertYearTipsByAutoRenew(autoRenew: boolean): string {
  return autoRenew ? '续期证书' : ''
}
