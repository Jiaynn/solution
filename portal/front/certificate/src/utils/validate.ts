/*
 * @file formstate validate
 * @author zhu hao <zhuhao@qiniu.com>
 */

import { standardDomainRegx, wildcardDomainRegx } from '../constants/domain'

export function validate(criteria: boolean, invalidMsg: string) {
  return criteria ? invalidMsg : null
}

export function notEmpty(value: unknown, name = '') {
  return validate(value == null || !value, `${name}不可以为空`)
}

export function inputValid(value: string) {
  return validate(/[`~～!！@#$%^&*_+<>?:"“”{},，。./;；'‘’[\]]/.test(value), '不得含有非法字符')
}

export function mustContainChinese(value: string, name = '') {
  return validate(!/[\u4e00-\u9fa5]/.test(value), `${name}应包含中文字符`)
}

export function postCodeValid(value: string) {
  return validate(!/^\d{5,6}(?!\d)$/.test(value), '邮编格式不正确')
}

export function areaCodeValid(value: string) {
  return validate(!/^((0\d{2,3}))$/.test(value), '区号格式不正确')
}

export function telephoneValid(value: string) {
  return validate(!/^(\d{7,8})(-(\d{3,}))?$/.test(value), '座机号格式不正确')
}

export function mobilephoneValid(value: string) {
  return validate(!/^\d{11}$/.test(value), '手机号格式不正确')
}

export function emailValid(value: string) {
  return validate(!/^\w+([-+.]\w+)*@\w+([-.]\w+)*\.\w+([-.]\w+)*$/.test(value), '邮箱格式不正确')
}

export function wildcardDomainValid(value: string) {
  return validate(!wildcardDomainRegx.test(value), '泛域名格式不正确')
}

export function standardDomainValid(value: string) {
  return validate(!standardDomainRegx.test(value), '标准域名格式不正确')
}

export function onlySupportChineseAndCharacter(value: string) {
  return validate(!/^[a-zA-Z0-9_\s\-*.()\u4e00-\u9fa5]+$/.test(value), '仅支持中英文字符、数字、空格、_、-、*、.、()')
}
