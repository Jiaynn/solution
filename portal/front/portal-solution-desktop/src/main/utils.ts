import { join } from 'path'
import { BrowserWindow, dialog, session, shell } from 'electron'
import cmd from 'node-cmd'

import { pageUrl } from './config'
import { ElectronBridgeApi } from '../preload'

import icon from '../../resources/logo.png?asset'

export const createWindow = (): BrowserWindow => {
  // Create the browser window.
  const minWidth = 1024
  const minHeight = 768
  const mainWindow = new BrowserWindow({
    width: minWidth,
    height: minHeight,
    minWidth,
    minHeight,
    show: false,
    autoHideMenuBar: true,
    titleBarStyle: 'hidden',
    ...(process.platform === 'linux' ? { icon } : {}),
    webPreferences: {
      preload: join(__dirname, '../preload/index.js'),
      sandbox: false,
      nodeIntegration: true
    }
  })

  mainWindow.center()

  mainWindow.on('ready-to-show', () => {
    mainWindow.show()
  })

  mainWindow.webContents.setWindowOpenHandler((details) => {
    shell.openExternal(details.url)
    return { action: 'deny' }
  })

  // HMR for renderer base on electron-vite cli.
  // Load the remote URL for development or the local html file for production.
  mainWindow.loadURL(pageUrl)

  return mainWindow
}

export const persistentCookie = (): void => {
  session.defaultSession.cookies.on('changed', (_, cookie) => {
    if (cookie.session && cookie.name === 'PORTAL_SESSION') {
      session.defaultSession.cookies.set({
        url: pageUrl,
        name: cookie.name,
        value: cookie.value,
        domain: cookie.domain,
        path: cookie.path,
        secure: cookie.secure,
        httpOnly: cookie.httpOnly,
        expirationDate: Math.floor(Date.now() / 1000) + 60 * 60 * 24 * 2,
        sameSite: cookie.sameSite
      })
    }
  })
}

type Platform = Parameters<ElectronBridgeApi['openEditor']>[0]['platform']
export const callEditor = (platform: Platform, dest: string): void => {
  if (process.platform === 'darwin') {
    // 运行在 macOS 上
    if (platform === 'Android') {
      cmd.run(`open -a /Applications/Android\\ Studio.app ${dest}`)
      return
    }
    if (platform === 'iOS') {
      cmd.run(`open -a /Applications/Xcode.app ${dest}`)
      return
    }
  }
  if (process.platform === 'win32') {
    // 运行在 Windows 上
    // ...
    return
  }
  if (process.platform === 'linux') {
    // 运行在 Linux 上
    // ...
  }
  dialog.showMessageBox({ message: '打开失败，请检查编辑器是否安装或项目是否存在' })
}
