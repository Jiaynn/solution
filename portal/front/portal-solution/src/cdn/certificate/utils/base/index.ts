/*
 * @file base utils
 * @author nighca <nighca@live.cn>
 */

import moment from 'moment'

import { RouterStore } from 'portal-base/common/router'

export interface IPayOrderOptions {
  orderId: string       // fusion 订单号
  tradeOrderId: string  // BO 订单号
}

export function payForOrder({ tradeOrderId }: IPayOrderOptions, routerStore: RouterStore) {
  routerStore.push(`/financial/verify-order/${tradeOrderId}`)
}

export function humanizeTime(time: number, format?: string) {
  return format
    ? moment(time * 1000).format(format)
    : moment(time * 1000).format('YYYY-MM-DD HH:mm:ss')
}
