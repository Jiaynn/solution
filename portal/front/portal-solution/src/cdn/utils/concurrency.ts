/**
 * @file concurrency limit request
 * @source https://gist.github.com/nighca/6562d098ac01814b6e1c1718b16d4abc
 */

export default function concurrency<T>(process: (_: any) => Promise<T>, limit = -1) {
  return function batchProcess(tasks: unknown[] = []): Promise<T[]> {
    const results: T[] = []
    let finished = 0
    let current = 0
    let rejected = false

    function tryProcess(resolve: (...args: any) => any, reject: (...args: any) => any) {
      if (rejected) return
      if (finished >= tasks.length) return resolve(results)
      if (current >= tasks.length) return

      const index = current++
      process(tasks[index]).then(
        result => {
          results[index] = result
          finished++
          tryProcess(resolve, reject)
        },
        err => {
          reject(err)
          rejected = true
        }
      )
    }

    return new Promise((resolve, reject) => {
      const realLimit = limit > 0 ? limit : tasks.length
      for (let i = 0; i < realLimit; i++) {
        tryProcess(resolve, reject)
      }
    })
  }
}
