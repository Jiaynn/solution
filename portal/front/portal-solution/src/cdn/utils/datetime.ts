import moment from 'moment'

export function isLastDayOfMonth() {
  return moment().endOf('month').format('MM-DD') === moment().format('MM-DD')
}
