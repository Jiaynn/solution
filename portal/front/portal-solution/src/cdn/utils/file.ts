export function saveFile(filename: string, content: string) {
  const blob = new Blob([content])
  const a = document.createElement('a')
  const url = window.URL.createObjectURL(blob)
  a.href = url
  a.download = filename
  a.click()
  window.URL.revokeObjectURL(url)
}
