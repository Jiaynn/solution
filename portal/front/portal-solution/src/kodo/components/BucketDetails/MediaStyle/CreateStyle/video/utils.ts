export enum SourceFormat {
  Mp4 = 'mp4',
  Flv = 'flv',
  Mov = 'mov',
}

export enum OutputFormat {
  Jpg = 'jpg',
  Png = 'png',
  Gif = 'gif'
}

export enum CoverType {
  Static = 'static',
  Dynamic = 'dynamic'
}

export const coverTypeNameMap = {
  [CoverType.Static]: '静态封面',
  [CoverType.Dynamic]: '动态封面'
}

export const coverTypes = [CoverType.Static] // [CoverType.Static, CoverType.Dynamic]
export const sourceFormatList = [SourceFormat.Mp4, SourceFormat.Flv, SourceFormat.Mov]
export const allowPreviewMimeTypeList = [
  'video/mp4',
  'video/x-flv',
  'video/quicktime'
]

export const allowPickAccept = {
  name: '视频',
  maxSize: 500 * 1024 * 1024,
  mimeTypes: allowPreviewMimeTypeList
}

export const outputFormatList = [OutputFormat.Jpg, OutputFormat.Png, OutputFormat.Gif]
export const staticCoverSuffixes = [OutputFormat.Jpg, OutputFormat.Png]
export const dynamicCoverSuffixes = [OutputFormat.Gif]

export enum CoverScaleType {
  None = 'none',
  // 限定宽, 高同比例缩放
  ScaleWidth = 'scaleWidth',
  // 限定高, 宽同比例缩放
  ScaleHeight = 'scaleHeight',
  // 限定宽高同比例缩放
  ScaleWidthHeight = 'scaleWidthHeight',
}

export const coverScaleTypes = [
  CoverScaleType.None,
  CoverScaleType.ScaleWidth,
  CoverScaleType.ScaleHeight,
  CoverScaleType.ScaleWidthHeight
]

export const coverScaleTypeNameMap = {
  [CoverScaleType.None]: '不缩放',
  [CoverScaleType.ScaleWidth]: '限定宽、高同比例缩放',
  [CoverScaleType.ScaleHeight]: '限定高、宽同比例缩放',
  [CoverScaleType.ScaleWidthHeight]: '限定宽高同比例缩放'
}

export enum CoverAutoScaleType {
  None = 0,
  One = 1,
  Two = 2
}

export const coverAutoScaleTypeNameMap = {
  [CoverAutoScaleType.None]: '不启用',
  [CoverAutoScaleType.One]: 1,
  [CoverAutoScaleType.Two]: 2
}

export const coverAutoScaleTypes = [CoverAutoScaleType.None, CoverAutoScaleType.One, CoverAutoScaleType.Two]

export function getCoverScaleType(w: number | null, h: number | null) {
  if (w == null && h == null) {
    return CoverScaleType.None
  }
  if (w != null && h == null) {
    return CoverScaleType.ScaleWidth
  }

  if (w == null && h != null) {
    return CoverScaleType.ScaleHeight
  }

  return CoverScaleType.ScaleWidthHeight
}

export function isValidSourceFormat(format: string): format is SourceFormat {
  return sourceFormatList.includes(format as any)
}

export function encodeKodoURI(bucketName: string, fileKey: string) {
  return `kodo://${bucketName}/${fileKey}`
}

export function decodeKodoURI(URI: string): [bucketName: string, fileKey: string] {
  const execArray = /kodo:\/\/([^/]+)\/(.*)/.exec(URI)
  if (execArray) {
    return execArray.slice(1, 3) as [string, string]
  }
  // 不符合规则时，直接把输入当作 fileKey 返回
  return ['', URI]
}
