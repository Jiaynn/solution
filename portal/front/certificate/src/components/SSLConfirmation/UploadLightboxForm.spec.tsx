/**
 * @file test cases for component UploadLightboxForm for SSLConfirmation
 * @author [Yao Jingtian] [yncst233@gmail.com]
 */
import React from 'react'

import { RendererUtils as Renderer } from '../../utils/test'
import UploadLightboxForm, { IUploadFormProps } from './UploadLightboxForm'

const props: IUploadFormProps = {
  orderid: '123',
  onCancel: () => { console.log('canceled') }
}

it('renders correctly', () => {
  const renderer = new Renderer()
  const tree = renderer.create(
    <UploadLightboxForm {...props} />
  ).toJSON()
  expect(tree).toMatchSnapshot()
})
