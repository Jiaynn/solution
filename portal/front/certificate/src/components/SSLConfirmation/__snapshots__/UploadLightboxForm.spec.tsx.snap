// Jest Snapshot v1, https://goo.gl/fbAQLP

exports[`renders correctly 1`] = `
<form
  className="ant-form ant-form-horizontal"
  onSubmit={[Function]}
>
  <div
    className="ant-row ant-form-item ssl-upload"
  >
    <div
      className="ant-col ant-form-item-control-wrapper"
    >
      <div
        className="ant-form-item-control"
      >
        <span
          className="ant-form-item-children"
        >
          <button
            className="ant-btn upload-btn"
            onClick={[Function]}
            type="button"
          >
            <i
              aria-label="icon: plus"
              className="anticon anticon-plus"
            >
              <svg
                aria-hidden="true"
                className=""
                data-icon="plus"
                fill="currentColor"
                focusable="false"
                height="1em"
                viewBox="64 64 896 896"
                width="1em"
              >
                <path
                  d="M482 152h60q8 0 8 8v704q0 8-8 8h-60q-8 0-8-8V160q0-8 8-8z"
                />
                <path
                  d="M176 474h672q8 0 8 8v60q0 8-8 8H176q-8 0-8-8v-60q0-8 8-8z"
                />
              </svg>
            </i>
            <span>
              选择文件
            </span>
            <input
              accept="image/*, application/pdf, application/zip"
              className="ant-input upload-input"
              disabled={false}
              id="file_data"
              multiple={true}
              name="file_data"
              onChange={[Function]}
              onKeyDown={[Function]}
              type="file"
              value=""
            />
          </button>
          <div
            className="file-name"
          />
          <p
            className="upload-notice"
          >
            不支持多文件上传，如果有多个文件，请压缩成一个文件后再上传。
          </p>
          <hr
            className="split-line sub-split-line"
          />
          <p
            className="help"
          >
            1. 上传文件类型可以是zip、jpg、jpeg、png、gif、tiff、pdf，单个文件大小不得大于5MB。
            <br />
            2. 请正确填写公司信息确认函并加盖两处公章(骑缝章＋末尾盖章)，再将扫描件上传提交进行人工审核。
            <br />
            3. 上传后请联系人保持联系方式(手机、邮箱)畅通以方便后续信息审核工作。
            <br />
          </p>
        </span>
      </div>
    </div>
  </div>
  <hr
    className="split-line sub-split-line"
  />
  <div
    className="footer-right"
  >
    <button
      className="ant-btn ant-btn-primary"
      disabled={true}
      onClick={[Function]}
      type="button"
    >
      <span>
        确定上传
      </span>
    </button>
    <button
      className="ant-btn"
      onClick={[Function]}
      type="button"
    >
      <span>
        取消
      </span>
    </button>
  </div>
</form>
`;
